$OpenBSD: patch-libavcodec_aacenc_utils_h,v 1.2 2015/09/23 09:16:45 ajacoutot Exp $

aacenc: copy PRNG from the decoder

AAC encoder: simplify and speed up find_min_book

--- libavcodec/aacenc_utils.h.orig	Mon Sep  7 21:58:01 2015
+++ libavcodec/aacenc_utils.h	Wed Sep 23 03:52:58 2015
@@ -89,13 +89,10 @@ static inline int find_min_book(float maxval, int sf)
     float Q34 = sqrtf(Q * sqrtf(Q));
     int qmaxval, cb;
     qmaxval = maxval * Q34 + C_QUANT;
-    if      (qmaxval ==  0) cb = 0;
-    else if (qmaxval ==  1) cb = 1;
-    else if (qmaxval ==  2) cb = 3;
-    else if (qmaxval <=  4) cb = 5;
-    else if (qmaxval <=  7) cb = 7;
-    else if (qmaxval <= 12) cb = 9;
-    else                    cb = 11;
+    if (qmaxval >= (FF_ARRAY_ELEMS(aac_maxval_cb)))
+        cb = 11;
+    else
+        cb = aac_maxval_cb[qmaxval];
     return cb;
 }
 
@@ -126,6 +123,15 @@ static inline int quant_array_idx(const float val, con
         }
     }
     return index;
+}
+
+/*
+ * linear congruential pseudorandom number generator, copied from the decoder
+ */
+static inline int lcg_random(unsigned previous_val)
+{
+    union { unsigned u; int s; } v = { previous_val * 1664525u + 1013904223 };
+    return v.s;
 }
 
 #define ERROR_IF(cond, ...) \
